scimono-server/src/main/java/com/sap/scimono/entity/User.java

```diff
@@ -620,7 +620,7 @@ public boolean equals(final Object obj) {
     if (!Objects.equals(title, other.title)) {
       return false;
     }
-    if (!Objects.equals(userName, other.userName)) {
+    if (!userName.equalsIgnoreCase(other.userName)) {
```

Comment:
If both values are null, this produces NPE. Keep the old check and combine it with the new one